### PR TITLE
Consolidate and reduce dependency footprint for iceberg extension

### PR DESCRIPTION
**Pull request by a2l007**
### Description

Reworked the dependencies for the iceberg extension to reduce the extension size and dependency footprint. Following is a summary of the changes:
- Replaced iceberg spark runtime dependency with `iceberg-core` and `iceberg-parquet` for tests
- Excluded `hive-serde` and replaced it with `hive-common` for runtime scope
- Bunch of transitive dependency exclusions.

<hr>

This PR has:

- [x] been self-reviewed.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: extensions-contrib/druid-iceberg-extensions/pom.xml [hunk 1]
```diff
@@ -35,6 +35,8 @@
   <modelVersion>4.0.0</modelVersion>
 
   <properties>
+    <iceberg.core.version>1.4.0</iceberg.core.version>
```
Comment (asdf2014) [inline extensions-contrib/druid-iceberg-extensions/pom.xml:38]:
```suggestion
    <iceberg.core.version>1.4.1</iceberg.core.version>
```

**Review by abhishekagarwal87 — APPROVED** [review]

**Comment by abhishekagarwal87** [comment]
@asdf2014 @a2l007  - ok to merge this?

File: extensions-contrib/druid-iceberg-extensions/pom.xml (hunk 1 above)
Comment (a2l007) [reply extensions-contrib/druid-iceberg-extensions/pom.xml:38]: I'm currently testing 1.4.1 and will be upgrading to it as part of another PR. Thanks.

Comment (asdf2014) [reply extensions-contrib/druid-iceberg-extensions/pom.xml:38]: @a2l007 No problem, thanks a lot

**Review by asdf2014 — APPROVED** [review]
LGTM 👍

**Pull request merged**